src/components/molecules/input/index.tsx

```diff
@@ -0,0 +1,160 @@
+import React, {
+  ChangeEventHandler,
+  FocusEventHandler,
+  MouseEventHandler,
+  useImperativeHandle,
+  useRef,
+} from "react"
+import MinusIcon from "../../fundamentals/icons/minus-icon"
+import PlusIcon from "../../fundamentals/icons/plus-icon"
+import InputContainer from "../../fundamentals/input-container"
+import InputHeader from "../../fundamentals/input-header"
+
+type inputProps = {
+  invalid?: boolean
+  placeholder?: string
+  defaultValue?: string
+  step?: string
+  min?: number
+  max?: number
+  inline?: boolean
+  label: string
+  name?: string
+  type?: string
+  inputStyle?: any
+  required?: boolean
+  value?: any
+  deletable?: boolean
+  onDelete?: MouseEventHandler<HTMLSpanElement>
+  onChange?: ChangeEventHandler<HTMLInputElement>
+  onFocus?: FocusEventHandler<HTMLInputElement>
+  textAlign?: string
+  disabled?: boolean
+  withTooltip?: boolean
+  tooltipText?: string
+  tooltipProps?: any
+  start?: any
+  props?: React.HTMLAttributes<HTMLDivElement>
+}
+
+const InputField = React.forwardRef(
+  (
+    {
+      invalid,
+      placeholder,
+      defaultValue,
+      step,
+      min,
+      max,
+      inline,
+      label,
+      name,
+      type,
```

Comment:
nit: curious to know why the props do not extend something like `InputHTMLAttributes<HTMLInputElement>`? This would allow the consumer of this component to pass in any html input attribute. We could use a rest parameter and pass it to the native `<input>` element. 

_PS: not 100% sure if the right type to extend is exactly `InputHTMLAttributes<HTMLInputElement>`_

Reply:
You are absolutely right, i have now made the props as so: 
```javascript 
type InputProps = React.InputHTMLAttributes<HTMLInputElement> & {
  label: string
  deletable?: boolean
  key?: string
  onDelete?: MouseEventHandler<HTMLSpanElement>
  onChange?: ChangeEventHandler<HTMLInputElement>
  onFocus?: FocusEventHandler<HTMLInputElement>
  withTooltip?: boolean
  tooltipText?: string
  tooltipProps?: any
  props?: React.HTMLAttributes<HTMLDivElement>
}
```

Where only the additional properties are added. Furthermore i then spread the rest of the props in the component and apply them to the field so we can pass all values existing on an `<input />`: 

```javascript
const InputField = React.forwardRef(
  (
    {
      placeholder,
      ...
      tooltipProps = {},
      props,
      ...fieldProps
    }: InputProps,
    ref
  ) => {
  ...
  return (
    ...
    <input
      ...
      {...fieldProps}
    />
    ...
  )
}
```